### PR TITLE
feat: ignore all event resource for gc

### PR DESCRIPTION
**Pull request by likakuli**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
ignore events in all groups for gc
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: pkg/controller/garbagecollector/graph_builder.go [hunk 1]
```diff
@@ -318,7 +321,9 @@ func (gb *GraphBuilder) Run(stopCh <-chan struct{}) {
 }
 
 var ignoredResources = map[schema.GroupResource]struct{}{
-	{Group: "", Resource: "events"}: {},
+	{Group: "", Resource: "events"}:                     {},
+	{Group: eventv1.GroupName, Resource: "events"}:      {},
+	{Group: eventv1beta1.GroupName, Resource: "events"}: {},
```
Comment (Abirdcfly) [inline pkg/controller/garbagecollector/graph_builder.go:326]: According to the statement here eventv1beta1 is deprecated in 1.22.So I guess v1 is enough.
https://github.com/kubernetes/kubernetes/blob/89aaf7c02dd33415ac994cecfddca7c02ed75834/staging/src/k8s.io/api/events/v1beta1/types.go#L27

**Comment by Abirdcfly** [comment]
I think it's more of a bug, not a feature.
By the way while looking at the code for this issue, I found another place with the same problem. https://github.com/kubernetes/kubernetes/pull/110939

File: pkg/controller/garbagecollector/graph_builder.go (hunk 1 above)
Comment (likakuli) [reply pkg/controller/garbagecollector/graph_builder.go]: ok, i have removed the v1beta1 version, thx

File: cmd/kube-controller-manager/app/options/options_test.go [hunk 2]
```diff
@@ -295,6 +297,7 @@ func TestAddFlags(t *testing.T) {
 				ConcurrentGCSyncs: 30,
 				GCIgnoredResources: []garbagecollectorconfig.GroupResource{
 					{Group: "", Resource: "events"},
+					{Group: eventv1.GroupName, Resource: "events"},
```
Comment (lavalamp) [inline cmd/kube-controller-manager/app/options/options_test.go:300]: Does the "" on the line above not act as a wildcard?

Comment (lavalamp) [reply cmd/kube-controller-manager/app/options/options_test.go:300]: hm maybe it doesn't. That is embarrassing to be wrong.

Comment (deads2k) [reply cmd/kube-controller-manager/app/options/options_test.go:300]: No, it doesn't.  it's the empty group.

Comment (lavalamp) [reply cmd/kube-controller-manager/app/options/options_test.go:300]: I don't even want to know how long it's been like this.

**Pull request merged**